docs/dev_docs.md

```diff
@@ -1,3 +1,449 @@
+
+```
+from pettingzoo.utils import to_parallel
+from pettingzoo.butterfly import pistonball_v3
+env = pistonball_v3.env()
+env = to_parallel(env)
+```
+
+#### Parallel to AEC
+
+Any parallel environment can be efficiently converted to an AEC environment with the `from_parallel` wrapper.
+
+```
+from pettingzoo.utils import from_parallel
+from pettingzoo.butterfly import pistonball_v3
+env = pistonball_v3.parallel_env()
+env = from_parallel(env)
+```
+
+### Utility wrappers
+
+We wanted our pettingzoo environments to be both easy to use and easy to implement. To combine these, we have a set of simple wrappers which provide input validation and other convenient reusable logic.
+
+* `BaseWrapper`: all AECEnv wrappers should inherit from this base class
+* `TerminateIllegalWrapper`: which handles illegal move logic for classic games
+* `CaptureStdoutWrapper`: which takes an environment which prints to terminal, and gives it an `ansi` render mode where it captures the terminal output and returns it as a string instead.
+* `AssertOutOfBoundsWrapper`: Asserts if the action given to step is outside of the action space. Applied in PettingZoo environments with discrete action spaces.
+* `ClipOutOfBoundsWrapper`: Clips the input action to fit in the continuous action space (emitting a warning if it does so). Applied to continuous environments in pettingzoo.
+* `OrderEnforcingWrapper`: Gives a sensible error message if function calls or attribute access are in a disallowed order, for example if step() is called before reset(), or .dones attribute is accessed before reset(), or if seed() and then step() is called before reset() is called again (reset must be called after seed()). Applied to all PettingZoo environments.
+
+All of the above can be applied to your environment with:
+
+```
+from pettingzoo.utils import OrderEnforcingWrapper
+from pettingzoo.butterfly import pistonball_v3
+env = pistonball_v3.env()
+env = OrderEnforcingWrapper(env)
+```
+
+## Developer Utils
+
+The utils directory contains a few functions which are helpful for debugging environments. These are documented in the API docs.
+
+The utils directory also contain some classes which are only helpful for developing new environments. These are documented below.
+
+### Agent selector
+
+The `agent_selector` class steps in a cycle
+
+It can be used as follows to cycle through the list of agents:
+
+```python
+from pettingzoo.utils import agent_selector
+agents = ["agent_1", "agent_2", "agent_3"]
+selector = agent_selector(agents)
+agent_selection = selector.reset()
+# agent_selection will be "agent_1"
+for i in range(100):
+    agent_selection = selector.next()
+    # will select "agent_2, agent_3, agent_1, agent_2, agent_3, ..."
+```
+
+### Depreciated Module
+
+The DepreciatedModule is used in PettingZoo to help guide the user away from old obsolete environment versions and to new ones. If you wish to create a similar versioning system, this may be helfpul.
```

Comment:
```suggestion
The DepreciatedModule is used in PettingZoo to help guide the user away from old obsolete environment versions and toward new ones. If you wish to create a similar versioning system, this may be helpful.
```